backend/infrahub/core/schema/attribute_schema.py

```diff
@@ -144,3 +183,33 @@ async def get_query_filter(
             partial_match=partial_match,
             support_profiles=support_profiles,
         )
+
+
+class TextAttributeSchema(AttributeSchema):
+    parameters: TextAttributeParameters = Field(
+        default_factory=TextAttributeParameters,
+        description="Extra parameters specific text attributes",
+        json_schema_extra={"update": UpdateSupport.VALIDATE_CONSTRAINT.value},
+    )
+
+    @model_validator(mode="after")
+    def reconcile_parameters(self) -> Self:
+        if self.regex != self.parameters.regex:
+            final_regex = self.parameters.regex or self.regex
+            self.regex = self.parameters.regex = final_regex
+        if self.min_length != self.parameters.min_length:
+            final_min_length = self.parameters.min_length or self.min_length
+            self.min_length = self.parameters.min_length = final_min_length
+        if self.max_length != self.parameters.max_length:
+            final_max_length = self.parameters.max_length or self.max_length
+            self.max_length = self.parameters.max_length = final_max_length
+        return self
```

Comment:
Is this something that we keep for some migration purposes or will we always have some duplicated data?

I think there are some subtle issues with this code with regards to `falsy` values, where it will work like this:

```python
min_length = None
param_min_length = 0
if min_length != param_min_length:
    final_min_length = param_min_length or min_length

type(min_length)
> <class 'NoneType'>
```

I'm not sure it will be relavant for text fields but if we add this for integers as the next step there would be a difference if you'd want to set `0` as the lowest number.

Reply:
It seems we allow users to specify both `max_length` and `parameters.max_length`. I think we should raise if users specify both instead of implicitly chosing `parameters.max_length`, as well as for other fields

Reply:
Also, considering we would set `max_length` with the value of `parameters.max_length`, could we get rid of `get_max_length` getter and keep using `max_length` ?